src/Element/ContactTabContent.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\contacts\Element;
+
+use Drupal\Core\Render\Element\RenderElement;
+
+/**
+ * Provides a dashboard tab content render element.
+ *
+ * Properties:
+ * - #tab: The tab entity being viewed.
+ * - #user: The user entity being viewed.
+ * - #subpage: The tab's dashboard subpage id.
+ * - #region_attributes: Attributes array for the content regions.
+ * - #content: Array of renderable regions.
+ *
+ * Usage example:
+ * @code
+ * $build['examples_tab_content'] = [
+ *   '#type' => 'contact_tab_content',
+ *   '#region_attributes' => [],
+ *   '#tab' => $tab,
+ *   '#subpage' => 'example',
+ *   '#user' => $user,
+ *   '#content' => [
+ *     'left' => [],
+ *     'right' => [],
+ *   ],
+ * ];
+ * @endcode
+ *
+ * @RenderElement("contact_tab_content")
+ */
+class ContactTabContent extends RenderElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    return [
+      '#theme' => 'contact_tab_content',
+      '#not_found' => $this->t('Page not found.'),
+      '#pre_render' => [
+        [$class, 'preRenderTabContent'],
+      ],
+    ];
+  }
+
+  /**
+   * Pre-render callback: Renders content of a tab.
+   *
+   * @param array $element
+   *   A structured array to build the tab content.
+   *
+   * @return array
+   *   The passed-in element containing the renderable regions in '#content'.
+   */
+  public static function preRenderTabContent(array $element) {
+    if ($element['#tab']) {
+      /* @var \Drupal\contacts\ContactsTabManager $tab_manager */
+      $tab_manager = \Drupal::service('contacts.tab_manager');
+      $blocks = $tab_manager->getBlocks($element['#tab'], $element['#user']);
```

Comment:
Let's pass this in as `#blocks`, as they are already compiled to verify the tab, unless we also want to move verification into here...

Reply:
We don't need to now that we are verifying as part of render in #64